corehq/apps/user_importer/importer.py

```diff
@@ -423,12 +423,14 @@ def _get_domain_info(domain):
                     user.add_phone_number(_fmt_phone(phone_number), default=True)
                 if name:
                     user.set_full_name(str(name))
-                if data:
-                    user.user_data.update(data)
+                data = data or {}
                 if profile:
-                    user.user_data[PROFILE_SLUG] = domain_info.profiles_by_name[profile].id
-                    for key in domain_info.profiles_by_name[profile].fields.keys():
-                        user.user_data.pop(key, None)   # TODO: pop fields when editing user, too
+                    data[PROFILE_SLUG] = domain_info.profiles_by_name[profile].id
+                if data:
```

Comment:
Does the metadata setter enforce this now?

Reply:
The TODO? Yes.